libtenzir/builtins/formats/json.cpp

```diff
@@ -151,6 +151,16 @@ truncate(std::string_view text, const size_t N = 50) -> std::string {
 /// Parses simdjson objects into the given `series_builder` handles.
 class doc_parser {
 public:
+  /// The result of a parsing operation
+  enum class result {
+    /// The parsing succeeded
+    success,
+    /// The parsing failed, but wrote elements into the builder
+    failure_with_write,
+    /// The parsing failed, but did not affect the builder
+    failure_no_change,
```

Comment:
When exactly would I get those? Only if the JSON itself is malformed in some way, or also in other cases?

Reply:
It will happen on any issue with the JSON, including non-structural issues, such as `1.2.3` as a number.